### PR TITLE
feat: use scikit-build-core

### PR DESCRIPTION
**Pull request by henryiii**
Moving to scikit-build-core.

### CODE REVIEW

**Comment by henryiii** [comment]
I'd like to make a follow-up dropping 2.7. Better if an SDist build is made, and better for dropping manylinux1. We could keep manylinux1 if we forced only binaries for cmake, I think, as it would then get the last manylinux1 cmake. But better just do update, I think.

**Comment by mayeut** [comment]
> We could keep manylinux1 if we forced only binaries for cmake, I think, as it would then get the last manylinux1 cmake. But better just do update, I think.

Should we build on manylinux2010 (as we do with CMake) ?
Maybe it's not worth it given the low number of downloads for manylinux 2010 (glibc 2.12) for ninja the last 30 days (I removed some rows in there):

| libc_version | cum_percent | download_count |
| ------------ | ----------: | -------------: |
| 2.35         |     32.88 % |      2,359,423 |
| 2.34         |     33.79 % |         74,313 |
| 2.31         |     72.24 % |      3,154,519 |
| 2.28         |     76.20 % |        324,935 |
| 2.27         |     80.20 % |        327,986 |
| 2.26         |     96.37 % |      1,326,990 |
| 2.24         |     96.39 % |          1,972 |
| 2.23         |     96.61 % |         17,466 |
| 2.17         |     99.95 % |        271,337 |
| 2.12         |    100.00 % |          4,214 |
| 2.9          |    100.00 % |             52 |
| 2.5          |    100.00 % |             41 |
|Total         |             |      8,206,430 |

**Comment by henryiii** [comment]
Not worth it, but fine to be consistent.

File: pyproject.toml [hunk 1]
```diff
@@ -1,53 +1,109 @@
...
-]
-before-build = "pip install -r requirements-repair.txt"
-repair-wheel-command = "python scripts/repair_wheel.py -w {dest_dir} {wheel}"
 test-extras = "test"
 test-command = "pytest {project}/tests"
 test-skip = ["*-win_arm64", "*-macosx_universal2:arm64"]
 environment-pass = ["SETUPTOOLS_SCM_PRETEND_VERSION"]
 
 [[tool.cibuildwheel.overrides]]
 select = "*-manylinux_{x86_64,i686}"
```
Comment (henryiii) [inline pyproject.toml:92]:
```suggestion
select = "*-manylinux_{x86_64,i686}"
manylinux-x86_64-image = "manylinux2010"
manylinux-i686-image = "manylinux2010"
```

**Comment by henryiii** [comment]
(Also when we drop 3.6 and below, that distribution may change)

**Review by mayeut — APPROVED** [review]

**Pull request merged**